cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -454,6 +471,40 @@ func generateMountsCommonInstallRecover(mst *initramfsMountsState) error {
 	return sysconfig.ConfigureTargetSystem(configOpts)
 }
 
+func isSavePresent() (bool, error) {
+	devPath := filepath.Join(dirs.GlobalRootDir, "/dev/disk/by-partlabel/ubuntu-save")
```

Comment:
I think this check is insufficient/wrong, we need to make sure that if ubuntu-save exists, it is on the same disk as the first partition we mounted using the `disk` from the first partition we mounted. The code in secbootUnlockVolumeIfEncrypted handles this correctly, so I think we should just skip this check entirely and use the return values from secbootUnlockVolumeIfEncrypted (or possibly adjust that function) to decide if there is an ubuntu-save present. It's slightly more complicated because it's valid if there's not an ubuntu-save partition (or are we now making ubuntu-save required for all uc20 installations?), but I still think that just checking udev partlabels like this is wrong.

Reply:
We do have a check a bit later, after the mount has completed, verifying that ubuntu-save comes from the same device as ubuntu-boot and thus ubuntu-data. Is that enough?

Reply:
Well that check is not enough, but the checks that happen inside secbootUnlockVolumeIfEncrypted to only mount ubuntu-save from the provided disk is sufficient to ensure we are not mounting untrusted external partitions which is unsafe. 

The attack that this check here exposes us to is that if we are booting a system without ubuntu-save (either encrypted or not - but see above question about whether ubuntu-save will be mandatory), and someone attaches a USB with an ubuntu-save, then secbootUnlockVolumeIfEncrypted will return an error which IIUC would break the boot process. We wouldn't be exposing ourselves to attack, but we would be enabling a DOS attack against the device which I think should be avoided if possible.